src/nemo_evaluator/solvers/harbor.py

```
@@ -974,6 +974,347 @@ def _split_key_for_tmux(key: str, max_escaped_len: int) -> list[str]:
+
+def _build_custom_tokenizer(spec: str) -> dict:
+    from litellm.utils import create_pretrained_tokenizer, create_tokenizer
+
+    spec_path = Path(spec)
+    if spec_path.is_file():
+        return create_tokenizer(spec_path.read_text())
+    return create_pretrained_tokenizer(spec)
+
+
+def _install_token_counter_patch() -> None:
+    global _TOKEN_COUNTER_PATCHED
```

Ruff (0.15.18)

[warning] Using the global statement to update `_TOKEN_COUNTER_PATCHED` is discouraged

(PLW0603)